applications/nrf5340_audio/src/modules/led.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "led.h"
+
+#include <zephyr.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+#include <hal/nrf_gpio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <device.h>
+
+#include "macros_common.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(led, CONFIG_LOG_LED_LEVEL);
+
+#define ACTIVE_LOW 1
+#define ACTIVE_HIGH 0
+
+#define BLINK_FREQ_MS 1000
+/* Maximum number of LED_UNITS. 1 RGB LED = 1 UNIT of 3 LEDS */
+#define LED_UNIT_MAX 10
+#define NUM_COLORS_RGB 3
+#define BASE_10 10
+
+#define CORE_APP_STR "CORE_APP"
+#define CORE_NET_STR "CORE_NET"
+
+#define LABEL_AND_COMMA(node_id) DT_LABEL(node_id),
+#define GPIO_AND_COMMA(node_id) DT_GPIO_PIN(node_id, gpios),
+#define GPIO_LABEL_AND_COMMMA(node_id) DT_GPIO_LABEL(node_id, gpios),
+#define GPIO_FLAGS_AND_COMMMA(node_id) DT_GPIO_FLAGS(node_id, gpios),
+
+/* The following arrays are populated compile time from the .dts*/
+const char *led_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), LABEL_AND_COMMA) };
+
+const int led_pins[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_AND_COMMA) };
+
+const char *led_gpio_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_LABEL_AND_COMMMA) };
+
+const int led_gpio_flags[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_FLAGS_AND_COMMMA) };
```

Comment:
The reason I didn't change this files was because the label is being used on line:151, so it would require a bigger rewrite. The plan is also to go over to using the dk_buttons_and_leds module in NCS later on

Reply:
I'm sorry but I don't think we have time to do that rewrite now, since the deadline for the merge into NCS is today. I will create a task for it in our backlog